core/util/runPuppet.js

```diff
@@ -71,114 +74,116 @@ async function processScenarioView (scenario, variantOrScenarioLabelSafe, scenar
+      } else {
+        console.warn('WARNING: script not found: ' + beforeScriptPath);
+      }
     }
-  }
 
-  //  --- OPEN URL ---
-  var url = translateUrl(scenario.url);
-  if (isReference && scenario.referenceUrl) {
-    url = scenario.referenceUrl;
-  }
-  await page.goto(url);
-
-  await injectBackstopTools(page);
-
-  //  --- WAIT FOR READY EVENT ---
-  var readyEvent = scenario.readyEvent || config.readyEvent;
-  if (readyEvent) {
-    await page.evaluate(`window._readyEvent = '${readyEvent}'`)
+    //  --- OPEN URL ---
+    var url = scenario.url;
+    if (isReference && scenario.referenceUrl) {
+      url = scenario.referenceUrl;
+    }
+    await page.goto(translateUrl(url));
+  
+    await injectBackstopTools(page);
 
-    await page.waitForFunction(() => {
-      return window._backstopTools.hasLogged(window._readyEvent);
-    });
+    //  --- WAIT FOR READY EVENT ---
+    var readyEvent = scenario.readyEvent || config.readyEvent;
+    if (readyEvent) {
+      await page.evaluate(`window._readyEvent = '${readyEvent}'`)
 
-    await page.evaluate(_ => console.info('readyEvent ok'));
-  }
+      await page.waitForFunction(() => {
+        return window._backstopTools.hasLogged(window._readyEvent);
+      });
 
-  // --- WAIT FOR SELECTOR ---
-  if (scenario.readySelector) {
-    await page.waitFor(scenario.readySelector);
-  }
-  //
+      await page.evaluate(_ => console.info('readyEvent ok'));
+    }
 
-  // --- DELAY ---
-  if (scenario.delay > 0) {
-    await page.waitFor(scenario.delay);
-  }
+    // --- WAIT FOR SELECTOR ---
+    if (scenario.readySelector) {
+      await page.waitFor(scenario.readySelector);
+    }
+    //
 
-  //--- REMOVE SELECTORS ---
-  if (scenario.hasOwnProperty('removeSelectors')) {
-    const removeSelectors = async () => {
-      return Promise.all(
-        scenario.removeSelectors.map(async (selector) => {
-          await page
-            .evaluate(`window._backstopSelector = '${selector}'`);
-
-          await page
-            .evaluate(() => {
-              Array.prototype.forEach.call(document.querySelectorAll(window._backstopSelector), function (s, j) {
-                s.style.display = 'none';
-                s.classList.add('__86d');
-              });
-            });
-        })
-      );
+    // --- DELAY ---
+    if (scenario.delay > 0) {
+      await page.waitFor(scenario.delay);
     }
 
-    await removeSelectors();
-  }
+    //--- REMOVE SELECTORS ---
+    if (scenario.hasOwnProperty('removeSelectors')) {
+      const removeSelectors = async () => {
+        return Promise.all(
+          scenario.removeSelectors.map(async (selector) => {
+            await page
+              .evaluate(`window._backstopSelector = '${selector}'`);
+
+            await page
+              .evaluate(() => {
+                Array.prototype.forEach.call(document.querySelectorAll(window._backstopSelector), function (s, j) {
```

Comment:
This could also be `Array.from(document.querySelectorAll(...).forEach()` its easier to understand.
`j` is also unused